mobile_config/src/sub_dao_service.rs

```diff
@@ -0,0 +1,95 @@
+use crate::{
+    key_cache::KeyCache, sub_dao_epoch_reward_info, telemetry, verify_public_key, GrpcResult,
+};
+use chrono::Utc;
+use file_store::traits::{MsgVerify, TimestampEncode};
+use helium_crypto::{Keypair, PublicKey, Sign};
+use helium_proto::{
+    services::sub_dao::{self, SubDaoEpochRewardInfoReqV1, SubDaoEpochRewardInfoResV1},
+    Message,
+};
+use sqlx::{Pool, Postgres};
+use std::sync::Arc;
+use tonic::{Request, Response, Status};
+
+pub struct SubDaoService {
+    key_cache: KeyCache,
+    metadata_pool: Pool<Postgres>,
+    signing_key: Arc<Keypair>,
+}
+
+impl SubDaoService {
+    pub fn new(key_cache: KeyCache, metadata_pool: Pool<Postgres>, signing_key: Keypair) -> Self {
+        Self {
+            key_cache,
+            metadata_pool,
+            signing_key: Arc::new(signing_key),
+        }
+    }
+
+    fn verify_request_signature<R>(&self, signer: &PublicKey, request: &R) -> Result<(), Status>
+    where
+        R: MsgVerify,
+    {
+        if self.key_cache.verify_signature(signer, request).is_ok() {
+            tracing::debug!(signer = signer.to_string(), "request authorized");
+            return Ok(());
+        }
+        Err(Status::permission_denied("unauthorized request signature"))
+    }
+
+    fn verify_request_signature_for_info(
+        &self,
+        request: &SubDaoEpochRewardInfoReqV1,
+    ) -> Result<(), Status> {
+        let signer = verify_public_key(&request.signer)?;
+        self.verify_request_signature(&signer, request)
+    }
+
+    fn sign_response(&self, response: &[u8]) -> Result<Vec<u8>, Status> {
+        self.signing_key
+            .sign(response)
+            .map_err(|_| Status::internal("response signing error"))
+    }
+}
+
+#[tonic::async_trait]
+impl sub_dao::sub_dao_server::SubDao for SubDaoService {
+    async fn info(
+        &self,
+        request: Request<SubDaoEpochRewardInfoReqV1>,
+    ) -> GrpcResult<SubDaoEpochRewardInfoResV1> {
+        let request = request.into_inner();
+        telemetry::count_request("sub_dao_reward_info", "info");
+        custom_tracing::record("sub_dao", &request.sub_dao_address);
```

Comment:
does this need a custom metric? there's only one subdao in the context of the mobile config service, no? and the read context for this is always going to be in the metrics pulled from the mobile config service/host so this shouldn't ever conflict with the metrics from the iot config service?

Reply:
correct, it wont conflict with the iot config service metric.